lib/dialects/db2/query.js

```diff
@@ -0,0 +1,508 @@
+'use strict';
+
+const AbstractQuery = require('../abstract/query');
+const sequelizeErrors = require('../../errors');
+const parserStore = require('../parserStore')('db2');
+const _ = require('lodash');
+const { logger } = require('../../utils/logger');
+const moment = require('moment');
+const debug = logger.debugContext('sql:db2');
+
+class Query extends AbstractQuery {
+  getInsertIdField() {
+    return 'id';
+  }
+
+  getSQLTypeFromJsType(value) {
+    const param = { ParamType: 'INPUT', Data: value };
+    if (Buffer.isBuffer(value)) {
+      param.DataType = 'BLOB';
+      return param;
+    }
+    return value;
+  }
+
+  async _run(connection, sql, parameters) {	
+    this.sql = sql;
+    //do we need benchmark for this query execution
```

Comment:
I would resolve this question before merging

Reply:
Done.